coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/CoralToTrinoSqlCallTransformersUtil.java

```diff
@@ -0,0 +1,247 @@
+    sqlCallTransformerList
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralOperator("array_contains"), 2, "contains"));
+    sqlCallTransformerList
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralOperator("regexp_extract"), 3, "regexp_extract",
+            "[{\"input\": 1}, {\"op\": \"hive_pattern_to_trino\", \"operands\":[{\"input\": 2}]}, {\"input\": 3}]",
+            null, null));
+    sqlCallTransformerList
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralOperator("instr"), 2, "strpos"));
+    sqlCallTransformerList.add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralOperator("decode"), 2,
+        "[{\"regex\":\"(?i)('utf-8')\", \"input\":2, \"name\":\"from_utf8\"}]", "[{\"input\":1}]", null, null));
+
+    sqlCallTransformerList
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralOperator("date_add"), 2, "date_add",
+            "[{\"value\": 'day'}, {\"input\": 2},  "
+                + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+            null, null));
+    sqlCallTransformerList
+        .add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralOperator("date_sub"), 2, "date_add",
+            "[{\"value\": 'day'}, " + "{\"op\": \"*\", \"operands\":[{\"input\": 2}, {\"value\": -1}]}, "
+                + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+            null, null));
+    sqlCallTransformerList.add(createSignatureBasedConditionSqlCallTransformer(hiveToCoralOperator("datediff"), 2,
+        "date_diff",
+        "[{\"value\": 'day'}, {\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 2}]}]}, "
+            + "{\"op\": \"date\", \"operands\":[{\"op\": \"timestamp\", \"operands\":[{\"input\": 1}]}]}]",
+        null, null));
+  }
+
+  private static void initializeLinkedInFunctionTransformers(List<SqlCallTransformer> sqlCallTransformerList) {
+    // Most "com.linkedin..." UDFs follow convention of having UDF names mapped from camel-cased name to snake-cased name.
+    // For example: For class name IsGuestMemberId, the conventional udf name would be is_guest_member_id.
+    // While this convention fits most UDFs it doesn't fit all. With the following mapping we override the conventional
+    // UDF name mapping behavior to a hardcoded one.
+    // For example instead of UserAgentParser getting mapped to user_agent_parser, we mapped it here to useragentparser
+    Set<String> linkedInFunctionIdSet = new HashSet<>();
+    sqlCallTransformerList.add(createSignatureBasedConditionSqlCallTransformer(
+        linkedInFunctionToCoralOperator("com.linkedin.dali.udf.watbotcrawlerlookup.hive.WATBotCrawlerLookup"), 3,
+        "wat_bot_crawler_lookup"));
+    addLinkedInFunctionSignature(linkedInFunctionIdSet,
+        "com.linkedin.dali.udf.watbotcrawlerlookup.hive.WATBotCrawlerLookup", 3);
```

Comment:
Let's combine them within one helper method like `addLinkedInUDFTransformer`, so we don't need to write a UDF name twice.

Reply:
Done